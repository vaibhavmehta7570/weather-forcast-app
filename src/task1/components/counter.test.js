import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import Counter from './counter'
import renderer from 'react-test-renderer'
configure({ adapter: new Adapter() })

describe('Test counter.jsx', () => {
  const wrapperApp = shallow(<Counter />)

  test('render correctly Counter component', () => {
    const appRenderResponse = renderer.create(<Counter />).toJSON()
    expect(appRenderResponse).toMatchSnapshot()
  })

  test('simulates onIncrement updation', () => {
    const value = wrapperApp.find('.counter-value').text()
    expect(value).toBe('0')
    const incrementbutton = wrapperApp.find('.btn-green')
    incrementbutton.simulate('click')
    const updatedValue = wrapperApp.find('.counter-value').text()
    expect(updatedValue).toBe('1')
  })

  test('simulates onReset updation', () => {
    const value = wrapperApp.find('.counter-value').text()
    expect(value).toBe('1')
    const resetButton = wrapperApp.find('.btn-orange')
    resetButton.simulate('click')
    const updatedValue = wrapperApp.find('.counter-value').text()
    expect(updatedValue).toBe('0')
  })

  test('simulates onDecrement updation', () => {
    const value = wrapperApp.find('.counter-value').text()
    expect(value).toBe('0')
    const decrementButton = wrapperApp.find('.btn-red')
    decrementButton.simulate('click')
    const updatedValue = wrapperApp.find('.counter-value').text()
    expect(updatedValue).toBe('-1')
  })
})
