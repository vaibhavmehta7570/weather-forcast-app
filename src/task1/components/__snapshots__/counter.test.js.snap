// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test counter.jsx render correctly Counter component 1`] = `
<div
  className="counter-app"
>
  <section
    className="counter-value"
  >
    0
  </section>
  <section
    className="icon-panel"
  >
    <button
      className="btn btn-green"
      onClick={[Function]}
    />
    <button
      className="btn btn-orange"
      onClick={[Function]}
    />
    <button
      className="btn btn-red"
      onClick={[Function]}
    />
  </section>
</div>
`;
